vendor/hal_camera_default.te

```diff
@@ -1,6 +1,9 @@
 vndbinder_use(hal_camera_default);
 allow hal_camera_default qdisplay_service:service_manager find;
 
+allow hal_camera_default self:socket { create ioctl read write };
```

Comment:
The ioctl still fails for me on tone. It wants  `0xc302` and `0xc304` for me, but only granting `allowxperm hal_camera_default self:socket { 0xc302 0xc304 };` still fails with:
```
avc: denied { ioctl } for path="socket:[23121]" dev="sockfs" ino=23121 ioctlcmd=0xc302  scontext=u:r:hal_camera_default:s0 tcontext=u:r:hal_camera_default:s0 tclass=socket
```
You cannot allow a general self:socket ioctl (because that would include `0x00008905` which is a neverallow).
I don't understand if specifying an allowxperm after allowing the general ioctl narrows it down again, please enlighten me

Reply:
Ok, this seems to work so far, seems both the general ioctl and the allowxperm are needed